src/Future.js

```diff
@@ -133,7 +133,7 @@ Future.cache = function(f) {
 
   function doResolve(reject, resolve) {
     status = 'PENDING';
-    return f.fork(
+    return f._fork(
```

Comment:
After some thinking and code-path tracing I realized handleCompletion does not deal with user-land functions that haven't been "jailed" already. So it should be safe to apply the optimization here too.